src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -121,15 +132,307 @@ export const CockpitStorageIntegration = ({
+    const mountPointConstraints = useMountPointConstraints();
+    const requiredSize = useRequiredSize();
+
+    const newMountPoints = useMemo(() => JSON.parse(window.sessionStorage.getItem("cockpit_mount_points") || "{}"), []);
+    const cockpitPassphrases = useMemo(() => JSON.parse(window.sessionStorage.getItem("cockpit_passphrases") || "{}"), []);
+
+    const useConfiguredStorage = useMemo(() => {
+        const availability = checkConfiguredStorage({
+            deviceData,
+            mountPointConstraints,
+            scenarioPartitioningMapping,
+            newMountPoints,
+        });
+
+        return availability.available;
+    }, [deviceData, mountPointConstraints, newMountPoints, scenarioPartitioningMapping]);
+
+    const useFreeSpace = useMemo(() => {
+        const availability = checkUseFreeSpace({ diskFreeSpace, diskTotalSpace, requiredSize });
+
+        return availability.available && !availability.hidden;
+    }, [diskFreeSpace, diskTotalSpace, requiredSize]);
+
+    const loading = !error && checkStep !== undefined;
+    const storageRequirementsNotMet = !loading && (error || (!useConfiguredStorage && !useFreeSpace));
+
+    useEffect(() => {
+        if (!useConfiguredStorage && checkStep === "prepare-partitioning") {
+            setCheckStep();
+        }
+    }, [useConfiguredStorage, checkStep]);
+
+    useEffect(() => {
+        if (checkStep !== "luks") {
+            return;
+        }
+
+        const devicesToUnlock = (
+            Object.keys(cockpitPassphrases)
+                    .map(dev => ({
+                        deviceName: deviceData[dev] ? dev : getDeviceNameByPath(deviceData, dev),
+                        passphrase: cockpitPassphrases[dev]
+                    })))
+                .filter(({ devicePath, deviceName }) => {
+                    return (
+                        deviceData[deviceName].formatData.type.v === "luks" &&
+                            deviceData[deviceName].formatData.attrs.v.has_key !== "True"
+                    );
+                });
+
+        if (devicesToUnlock.some(dev => !dev.passphrase)) {
+            onCritFail()({ message: _("Cockpit storage did not provide the passphrase to unlock encrypted device.") });
+        }
+
+        if (devicesToUnlock.length === 0) {
+            setCheckStep("prepare-partitioning");
+            return;
+        }
+
+        Promise.all(devicesToUnlock.map(unlockDevice))
+                .catch(exc => {
+                    setCheckStep();
+                    setError(exc);
+                })
+                .then(() => {
+                    dispatch(getDevicesAction());
+                });
+    }, [dispatch, checkStep, cockpitPassphrases, newMountPoints, deviceData, onCritFail, setError]);
+
+    useEffect(() => {
+        // If the required devices needed for manual partitioning are set up,
+        // prepare the partitioning and set the scenario to 'use-configured-storage'.
```

Comment:
I can't find the "set the scenario to 'use-configured-storage' part.

Reply:
Indeed, code was refactored. removed.